pilot/pkg/xds/eds.go

```diff
@@ -456,15 +456,18 @@ func shouldUseDeltaEds(req *model.PushRequest) bool {
 // onlyEndpointsChanged checks if a request contains *only* endpoints updates. This allows us to perform more efficient pushes
 // where we only update the endpoints that did change.
 func onlyEndpointsChanged(req *model.PushRequest) bool {
-	if len(req.ConfigsUpdated) > 0 {
-		for k := range req.ConfigsUpdated {
-			if k.Kind != kind.ServiceEntry {
-				return false
-			}
+	if len(req.ConfigsUpdated) == 0 {
```

Comment:
Could you add a small comment on top of this line explaining why ? I assume it's correct or no-regression, since this was the logic before - but I can't remember why no ConfigUpdates means something else than EP changed.

Reply:
`len(req.ConfigsUpdated) == 0` basically means "we don't know what configs are updated, just push anyways"

It happens like when:

- istiod responds to a new connected proxy
- updating MeshConfig triggers a global update
- ...